hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/client/utils/SparkValidatorUtils.java

```diff
@@ -116,16 +119,34 @@ private static CompletableFuture<Boolean> runValidatorAsync(SparkPreCommitValida
   /**
    * Get records from partitions modified as a dataset.
    * Note that this only works for COW tables.
+   *
+   * @param sqlContext          Spark {@link SQLContext} instance.
+   * @param partitionsAffected  A set of affected partitions.
+   * @param table               {@link HoodieTable} instance.
+   * @param newStructTypeSchema The {@link StructType} schema from after state.
+   * @return The records in Dataframe from committed files.
    */
   public static Dataset<Row> getRecordsFromCommittedFiles(SQLContext sqlContext,
-                                                          Set<String> partitionsAffected, HoodieTable table) {
-
+                                                          Set<String> partitionsAffected,
+                                                          HoodieTable table,
+                                                          StructType newStructTypeSchema) {
     List<String> committedFiles = partitionsAffected.stream()
         .flatMap(partition -> table.getBaseFileOnlyView().getLatestBaseFiles(partition).map(BaseFile::getPath))
         .collect(Collectors.toList());
 
     if (committedFiles.isEmpty()) {
-      return sqlContext.emptyDataFrame();
+      try {
+        return sqlContext.createDataFrame(
+            sqlContext.emptyDataFrame().rdd(),
```

Comment:
Can we return `Option<Dataset<Row>>`, there is no need to do any distinction operation when the before data set is empty.

Reply:
We still want to return a `Dataset<Row>` with the schema here to run the SQL query; otherwise, we cannot compare the SQL results between the before and after states.